graphql/util.lua

```diff
@@ -38,6 +47,20 @@ function util.coerceValue(node, schemaType, variables)
     return variables[node.name.value]
   end
 
+  if schemaType.__type == 'Union' then
+    local matched, result
+    for k, v in ipairs(schemaType.types) do
+        matched, result = pcall(util.coerceValue, node, v, variables)
+        if matched then
+          return result
+        end
+    end
+
+    local valid_types = util.map(schemaType.types, function (t) return t.name end)
+    --TODO! how to better report this error?
+    error('Could not coerce "' .. tostring(node.kind) .. '" to one of union types ['..table.concat(valid_types, ",")..']')
+  end
```

Comment:
seems the spec does not allow for arguments to be of "union" type.
http://facebook.github.io/graphql/#sec-Input-Values
This whole direction was started by the fact that i expected to be able to define an argument to be of type union, for example ```id: 10``` and ```id: "10"``` to be possible as long as i defined id argument of type union (my usecase was a bit more complicated but the idea is the same).
I guess i'll first ask the question on the language spec repo